src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -772,7 +773,7 @@ void SpatialPooler::updatePermanencesForColumn_(vector<Real>& perm,
   }
 
   numConnected = 0;
-  for (UInt i = 0; i < perm.size(); ++i)
+  for (UInt i = 0; i < perm.size(); i++)
```

Comment:
The most common advice is to stick with `++i` in for loops. Here's some more info:
http://stackoverflow.com/a/24858